src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -0,0 +1,334 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Parameters returned by the WWW-Authenticate header. This allows for dynamic
+    /// scenarios such as claim challenge, CAE, CA auth context.
+    /// See https://aka.ms/msal-net/wwwAuthenticate.
+    /// </summary>
+    public class WwwAuthenticateParameters
+    {
+        /// <summary>
+        /// Resource for which to request scopes.
+        /// This is the App ID URI of the API that returned the WWW-Authenticate header.
+        /// </summary>
+        public string Resource { get; set; }
+
+        /// <summary>
+        /// Scopes to request.
+        /// If it's not provided by the web API, it's computed from the Resource.
+        /// </summary>
+        public IEnumerable<string> Scopes
+        {
+            get
+            {
+                if (_scopes != null)
+                {
+                    return _scopes;
+                }
+                else if (!string.IsNullOrEmpty(Resource))
+                {
+                    return new[] { $"{Resource}/.default" };
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            set
+            {
+                _scopes = value;
+            }
+        }
+
+        private IEnumerable<string> _scopes;
+
+        /// <summary>
+        /// Authority from which to request an access token.
+        /// </summary>
+        public string Authority { get; set; }
+
+        /// <summary>
+        /// Claims demanded by the web API.
+        /// </summary>
+        public string Claims { get; set; }
+
+        /// <summary>
+        /// Error.
+        /// </summary>
+        public string Error { get; set; }
+
+        /// <summary>
+        /// Dictionary of raw parameters in the WWW-Authenticate header (extracted from the WWW-Authenticate header
+        /// string value, without any processing). This allows support for APIs which are not mappable easily to the standard
+        /// or framework specific (SAL, Microsoft.Identity.Web).
+        /// </summary>
+        public IDictionary<string, string> RawParameters { get; private set; }
```

Comment:
>RawParameters [](http://example.com/codeflow?start=43&length=13)

did you intent to make this public? You have the index'er below so not sure why this one would be public?

Reply:
The indexer came later, but i'll let JM decide.

Reply:
yeah, made internal.